pkg/apis/serving/v1/revision_lifecycle_test.go

```diff
@@ -690,33 +691,50 @@ func TestPropagateAutoscalerStatusReplicas(t *testing.T) {
 	testCases := []struct {
 		name                string
 		ps                  autoscalingv1alpha1.PodAutoscalerStatus
-		wantActualReplicas  int32
-		wantDesiredReplicas int32
+		wantActualReplicas  *int32
+		wantDesiredReplicas *int32
 	}{{
 		name: "active PodAutoScaler",
 		ps: autoscalingv1alpha1.PodAutoscalerStatus{
 			ActualScale:  ptr.Int32(1),
 			DesiredScale: ptr.Int32(2),
 		},
-		wantActualReplicas:  1,
-		wantDesiredReplicas: 2,
+		wantActualReplicas:  ptr.Int32(1),
+		wantDesiredReplicas: ptr.Int32(2),
 	}, {
 		name: "inactive PodAutoScaler",
 		ps: autoscalingv1alpha1.PodAutoscalerStatus{
 			DesiredScale: ptr.Int32(-1),
 		},
-		wantActualReplicas:  0,
-		wantDesiredReplicas: 0,
+		wantActualReplicas:  nil,
+		wantDesiredReplicas: nil,
 	}}
 
 	for _, tc := range testCases {
 		t.Run(tc.name, func(t *testing.T) {
 			r.PropagateAutoscalerStatus(&tc.ps)
-			if r.ActualReplicas != tc.wantActualReplicas {
-				t.Errorf("Expected r.ActualReplicas to be %d but got %d", tc.wantActualReplicas, r.ActualReplicas)
+
+			wantActual, gotActual := "nil", "nil"
+			if tc.wantActualReplicas != nil {
+				wantActual = strconv.Itoa(int(*tc.wantActualReplicas))
+			}
+			if r.ActualReplicas != nil {
+				gotActual = strconv.Itoa(int(*r.ActualReplicas))
 			}
-			if r.DesiredReplicas != tc.wantDesiredReplicas {
-				t.Errorf("Expected r.DesiredReplicas to be %d but got %d", tc.wantDesiredReplicas, r.DesiredReplicas)
+			if wantActual != gotActual {
+				t.Errorf("Expected r.ActualReplicas to be %s but got %s", wantActual, gotActual)
+			}
+
+			wantDesired, gotDesired := "nil", "nil"
+			if tc.wantDesiredReplicas != nil {
+				wantDesired = strconv.Itoa(int(*tc.wantDesiredReplicas))
+			}
+			if r.DesiredReplicas != nil {
+				gotDesired = strconv.Itoa(int(*r.DesiredReplicas))
+			}
+
+			if wantDesired != gotDesired {
+				t.Errorf("Expected r.DesiredReplicas to be %s but got %s", wantDesired, gotDesired)
```

Comment:
Total nit, but this can be done a bit more simply by using `cmp.Equal` and `cmp.Diff`.

```golang
if !cmp.Equal(tc.wantActualReplicas, r.ActualReplicas) {
	t.Errorf("r.ActualReplicas replicas wasn't as expected, (-want, +got):\n%s", cmp.Diff(tc.wantActualReplicas, r.ActualReplicas))
}

if !cmp.Equal(tc.wantDesiredReplicas, r.DesiredReplicas) {
	t.Errorf("r.DesiredReplicas replicas wasn't as expected, (-want, +got):\n%s", cmp.Diff(tc.wantDesiredReplicas, r.DesiredReplicas))
}
```

On error, this code produces something like:

```
--- FAIL: TestPropagateAutoscalerStatusReplicas (0.00s)
    --- FAIL: TestPropagateAutoscalerStatusReplicas/inactive_PodAutoScaler (0.00s)
        /home/mthoemme/go/src/knative.dev/serving/pkg/apis/serving/v1/revision_lifecycle_test.go:721: r.DesiredReplicas replicas wasn't as expected, (-want, +got):
              (*int32)(
            - 	nil,
            + 	&2,
              )
FAIL
FAIL	knative.dev/serving/pkg/apis/serving/v1	0.004s
FAIL
```

Reply:
OMG thanks! That's so much nicer!